odoo/tests/common.py

```diff
@@ -1960,17 +1991,77 @@ def setUp(self):
 
         self._logger = self._logger.getChild(self._testMethodName)
 
-        self.xmlrpc_common = xmlrpclib.ServerProxy(self.xmlrpc_url + 'common', transport=Transport(self.cr))
-        self.xmlrpc_db = xmlrpclib.ServerProxy(self.xmlrpc_url + 'db', transport=Transport(self.cr))
-        self.xmlrpc_object = xmlrpclib.ServerProxy(self.xmlrpc_url + 'object', transport=Transport(self.cr), use_datetime=True)
+        _self = self
+        class HttpCaseXmlrpcProxy(xmlrpclib.ServerProxy):
+            # ServerProxy's internal uses some weird magic..
+            def _ServerProxy__request(self, *args, **kwargs):
+                with _self.allow_requests(all_requests=True):
+                    return super()._ServerProxy__request(*args, **kwargs)
+
+        self.xmlrpc_common = HttpCaseXmlrpcProxy(self.xmlrpc_url + 'common', transport=Transport(self.cr))
+        self.xmlrpc_db = HttpCaseXmlrpcProxy(self.xmlrpc_url + 'db', transport=Transport(self.cr))
+        self.xmlrpc_object = HttpCaseXmlrpcProxy(self.xmlrpc_url + 'object', transport=Transport(self.cr), use_datetime=True)
         # setup an url opener helper
         self.opener = Opener(self.cr)
+        # we need to allow requests during pdf rendering.
+        old_run_wkhtmltopdf = ir_actions_report._run_wkhtmltopdf
+        def _patched_run_wkhtmltopdf(*args):
+            with self.allow_requests(all_requests=True):
+                return old_run_wkhtmltopdf(*args)
+        self.startPatcher(
+            patch.object(ir_actions_report, '_run_wkhtmltopdf', _patched_run_wkhtmltopdf)
+        )
```

Comment:
This creates a discrepancy between TransactionCase and HttpCase where the former requires an explicit `allow_pdf_rendering` while the latter does not, this however is in line with what was done before where pdf rendering required an explicit registry test mode while HttpCase enters it implicitly.